src/App.js

```diff
@@ -1,19 +1,37 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import { Provider } from 'react-redux';
+import store from './store/index';
+import GifPage from './components/GifPage/GifPage'
+import GifFavorite from './components/GifFavorite/GifFavorite';
+import { Route, NavLink, Switch, Redirect } from 'react-router-dom';
 
 class App extends Component {
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
+      <Provider store = {store}>
+      <div>   
+            <header>
+                <nav>
+                    <ul>
+                        <li><NavLink 
+                        to = "/home"
+                        exact>
+                        Home </NavLink></li>
+                        <li><NavLink 
+                        to = "/favorite"
+                        exact>
+                        Favorites</NavLink></li>
+                    </ul>
+                </nav>
+            </header>
+            <Switch>
+              <Route path = "/home" exact component = {GifPage} />
+              <Route path = "/favorite" exact component = {GifFavorite} />
+              <Redirect from = "/" to = "/home" />
```

Comment:
It is a good idea to have a redirect from every other non existing route point to a NotFound or to the index with a message.